### PR TITLE
Fix failing PHPUnit tests

### LINKED ISSUE #45254: PHP unit tests are failing
### Description

First reported [in Slack](https://wordpress.slack.com/archives/C02QB2JS7/p1666634913412629).

PHP unit tests in the Gutenberg repo [have been failing](https://github.com/WordPress/gutenberg/actions/runs/3313497602/jobs/5471558833) since [some time after 13:56 UTC and before 14:20 UTC](https://github.com/WordPress/gutenberg/commits/trunk). Unfortunately, this seems to point at my own [core commit](https://github.com/WordPress/wordpress-develop/commit/d962be18b5b1cf460556e4b40edd5192c524c637) (from 14:14 UTC)

The failure is

### Step-by-step reproduction instructions

Check any commit on GB's `trunk` that was merged after 2022-10-24 14:14 UTC.

Its PHP unit tests are failing with

```
There was 1 error:

1) WP_Navigation_Page_Test::test_gutenberg_navigation_init_function_generates_correct_preload_paths
Array to string conversion

/var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.0/client-assets.php:76
/var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.0/client-assets.php:109
/var/www/html/wp-includes/class-wp-hook.php:310
/var/www/html/wp-includes/plugin.php:205
/var/www/html/wp-includes/block-editor.php:[53](https://github.com/WordPress/gutenberg/actions/runs/3313985686/jobs/5472682507#step:8:54)9
/var/www/html/wp-content/plugins/gutenberg/lib/experimental/navigation-page.php:109
/var/www/html/wp-content/plugins/gutenberg/phpunit/navigation-page-test.php:73
phpvfscomposer:///var/www/html/wp-content/plugins/gutenberg/vendor/phpunit/phpunit/phpunit:97
```

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Resolves #45254.

PR tries to fix failing PHPUnit tests.

## How?
I've added a check to avoid calling the `gutenberg_resolve_assets` for WP 6.0 and above. The `__unstableResolvedAssets` will be generated using `_wp_get_iframed_editor_assets` for those versions.

## Testing Instructions
CI should be green.

Locally: `npm run test:unit:php -- --filter WP_Navigation_Page_Test`

### CODE REVIEW

**Comment by ockham** [comment]
Thanks @Mamaduka, looks like a great solution (that didn't occur to me). I'll quickly test locally to verify that it really falls back to `_wp_get_iframed_editor_assets` 😄

File: lib/compat/wordpress-6.0/client-assets.php [hunk 1]
```diff
@@ -105,6 +105,10 @@ function gutenberg_resolve_assets() {
 add_filter(
 	'block_editor_settings_all',
 	function( $settings ) {
+		if ( function_exists( '_wp_get_iframed_editor_assets' ) ) {
+			return $settings;
+		}
```
Comment (ockham) [inline lib/compat/wordpress-6.0/client-assets.php:111]: Let's maybe add a bit of an explanation here? Something like

```suggestion
		// The `__unstableResolvedAssets` are generated by `_wp_get_iframed_editor_assets` for WP >= 6.0.
		if ( function_exists( '_wp_get_iframed_editor_assets' ) ) {
			return $settings;
		}
```

**Comment by Mamaduka** [comment]
@ockham, I got inspiration from [your comment](https://github.com/WordPress/gutenberg/issues/45254#issuecomment-1289444224) for this fix 😄

**Comment by Mamaduka** [comment]
@ockham, I got inspiration from [your comment](https://github.com/WordPress/gutenberg/issues/45254#issuecomment-1289444224) for this fix 😄

**Review by ockham — APPROVED** [review]
LGTM! Thank you for fixing 😊

**Comment by ockham** [comment]
@priethor @juanmaguitar This should also be included in GB 14.4 please 😊

**Review by tyxla — APPROVED** [review]
Thank you @Mamaduka 😍

**Comment by ockham** [comment]
I just cherry-picked this PR to the wp/6.1 branch to get it included in the next release: 739b2b09c3

**Comment by ockham** [comment]
Cherry-picked to the [`release/14.4` branch](https://github.com/WordPress/gutenberg/commits/release/14.4) in e4b2cd2ea232f40ea1cf85575039889d2e95d459.

**Pull request merged**